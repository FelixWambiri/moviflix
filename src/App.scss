.App {
  text-align: center;
  display: flex;
  flex-direction: column;
  margin: auto;
  max-width: 1140px;
}

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

header {
  display: flex;
  margin: 20px;
  right: 5%;
  position: fixed;
}
