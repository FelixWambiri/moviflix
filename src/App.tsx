import React, { Component } from "react";
import "./App.scss";

/** Third party libraries**/
import { Provider } from "mobx-react";

/** Componets **/
import Header from "./components/Header";
import Episodes from "./components/Episodes";

/**Utility Fucntion **/
import { setUpMovieStore } from "./api/setup";
interface Props {}
interface State {
  movieStore: any;
}

export default class App extends Component<Props, State> {
  state = {
    movieStore: null
  };

  componentDidMount = (): void => {
    const { movieStore } = setUpMovieStore();
    this.setState({ movieStore });
  };
  render() {
    const { movieStore } = this.state;
    if (!movieStore) return null;

    return (
      <Provider movieStore={movieStore}>
        <div className="App">
          <Header />
          <Episodes />
        </div>
      </Provider>
    );
  }
}
