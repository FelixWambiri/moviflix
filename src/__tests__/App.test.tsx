import React from "react";
import { render, cleanup } from "@testing-library/react";

import App from "../App";

afterEach(cleanup);

describe("<App/>", () => {
  function renderAppComponent() {
    return render(<App />);
  }
  test("should match snapshot", () => {
    const {asFragment} = renderAppComponent()
    expect(asFragment()).toMatchSnapshot()
  });
});
