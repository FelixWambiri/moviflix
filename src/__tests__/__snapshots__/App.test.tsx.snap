// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header>
      <form
        data-testid="search-form"
      >
        <div
          class="search-box"
        >
          <input
            autocomplete="off"
            class="search-txt"
            data-testid="searchTerm"
            name="searchTerm"
            placeholder="Type to Search"
            type="text"
            value=""
          />
          <div
            class="search-button"
            data-testid="submit"
          >
            <i
              fill="currentColor"
              style="display: inline-block;"
            >
              <svg
                fill="currentColor"
                height="16"
                style="display: inline-block; vertical-align: middle;"
                viewBox="0 0 1664 1792"
                width="16"
              >
                <path
                  d="M1152 832q0-185-131.5-316.5t-316.5-131.5-316.5 131.5-131.5 316.5 131.5 316.5 316.5 131.5 316.5-131.5 131.5-316.5zM1664 1664q0 52-38 90t-90 38q-54 0-90-38l-343-342q-179 124-399 124-143 0-273.5-55.5t-225-150-150-225-55.5-273.5 55.5-273.5 150-225 225-150 273.5-55.5 273.5 55.5 225 150 150 225 55.5 273.5q0 220-124 399l343 343q37 37 37 90z"
                />
              </svg>
            </i>
          </div>
        </div>
      </form>
    </header>
    <div
      class="episodes-container"
    >
      <div>
        Search for a series
      </div>
    </div>
  </div>
</DocumentFragment>
`;
